File: CS/SpreadWinDemoCS/selection/selectionpolicy.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SpreadWinDemo.selection
{
    public partial class selectionpolicy : SpreadWinDemo.DemoBase
    {
        public selectionpolicy()
        {
            InitializeComponent();

            // シートの設定
            InitSheet(fpSpread1.Sheets[0]);

            comboBox1.SelectedIndexChanged += new EventHandler(comboBox1_SelectedIndexChanged);
        }

        private void InitSheet(FarPoint.Win.Spread.SheetView sheet)
        {
            //データ連結
            DataSet ds = new DataSet();
            ds.ReadXml(this.GetType().Assembly.GetManifestResourceStream(System.Reflection.Assembly.GetExecutingAssembly().GetName().Name + ".SampleData.data.xml"));
            sheet.DataSource = ds;

            //列幅の設定
            sheet.Columns[0].Width = 45;
            sheet.Columns[1].Width = 85;
            sheet.Columns[2].Width = 120;
            sheet.Columns[3].Width = 80;
            sheet.Columns[4].Width = 50;
            sheet.Columns[5].Width = 50;
            sheet.Columns[6].Width = 65;
            sheet.Columns[7].Width = 80;
            sheet.Columns[8].Width = 140;

            // 選択スタイルの設定
            sheet.SelectionStyle = FarPoint.Win.Spread.SelectionStyles.SelectionColors;
            sheet.SelectionBackColor = System.Drawing.Color.Red;
        }

        void comboBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            // 選択のクリア
            fpSpread1.Sheets[0].Models.Selection.ClearSelection();

            string[] strCom = comboBox1.Text.Split('.');

            switch (Convert.ToInt32(strCom[0]))
            {
                case 0:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.None;
                    break;
                case 1:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Cells;
                    break;
                case 2:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Rows;
                    break;
                case 3:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Rows | FarPoint.Win.Spread.SelectionBlockOptions.Cells;
                    break;
                case 4:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns;
                    break;
                case 5:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns | FarPoint.Win.Spread.SelectionBlockOptions.Cells;
                    break;
                case 6:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns | FarPoint.Win.Spread.SelectionBlockOptions.Rows;
                    break;
                case 7:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns | FarPoint.Win.Spread.SelectionBlockOptions.Rows | FarPoint.Win.Spread.SelectionBlockOptions.Cells;
                    break;
                case 8:
                    fpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Sheet;
                    break;
            }
        }


    }
}

File: VB/SpreadWinDemo/selection/selectionpolicy.vb
Public Class selectionpolicy

    Public Sub New()

        InitializeComponent()

        ' シートの設定
        InitSheet(FpSpread1.Sheets(0))

        ComboBox1.Text = "9.デフォルト"
    End Sub

    Private Sub InitSheet(sheet As FarPoint.Win.Spread.SheetView)
        'データ連結
        Dim ds As New DataSet()
        ds.ReadXml(Me.GetType().Assembly.GetManifestResourceStream(System.Reflection.Assembly.GetExecutingAssembly().GetName().Name & ".data.xml"))
        sheet.DataSource = ds

        '列幅の設定
        sheet.Columns(0).Width = 45
        sheet.Columns(1).Width = 85
        sheet.Columns(2).Width = 120
        sheet.Columns(3).Width = 80
        sheet.Columns(4).Width = 50
        sheet.Columns(5).Width = 50
        sheet.Columns(6).Width = 65
        sheet.Columns(7).Width = 80
        sheet.Columns(8).Width = 140

        ' 選択スタイルの設定
        sheet.SelectionStyle = FarPoint.Win.Spread.SelectionStyles.SelectionColors
        sheet.SelectionBackColor = System.Drawing.Color.Red
    End Sub

    Private Sub ComboBox1_SelectedIndexChanged(sender As Object, e As System.EventArgs) Handles ComboBox1.SelectedIndexChanged
        ' 選択のクリア
        FpSpread1.Sheets(0).Models.Selection.ClearSelection()

        Dim strCom As String() = ComboBox1.Text.Split("."c)

        Select Case Convert.ToInt32(strCom(0))
            Case 0
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.None
                Exit Select
            Case 1
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Cells
                Exit Select
            Case 2
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Rows
                Exit Select
            Case 3
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Rows Or FarPoint.Win.Spread.SelectionBlockOptions.Cells
                Exit Select
            Case 4
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns
                Exit Select
            Case 5
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns Or FarPoint.Win.Spread.SelectionBlockOptions.Cells
                Exit Select
            Case 6
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns Or FarPoint.Win.Spread.SelectionBlockOptions.Rows
                Exit Select
            Case 7
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Columns Or FarPoint.Win.Spread.SelectionBlockOptions.Rows Or FarPoint.Win.Spread.SelectionBlockOptions.Cells
                Exit Select
            Case 8
                FpSpread1.SelectionBlockOptions = FarPoint.Win.Spread.SelectionBlockOptions.Sheet
                Exit Select
        End Select
    End Sub
End Class
